dashboard/tasks_test.py

```diff
@@ -49,17 +49,14 @@ def test_celery_task_works(self):
         self.assertTrue(batch_update_user_data.delay())
 
     @override_settings(CELERY_ALWAYS_EAGER=True)
-    @patch('dashboard.api.get_student_current_grades', autospec=True)
-    @patch('dashboard.api.get_student_enrollments', autospec=True)
-    @patch('dashboard.api.get_student_certificates', autospec=True)
+    @patch('dashboard.api_edx_cache.CachedEdxUserData.update_cache_if_expired', new_callable=MagicMock)
     @patch('backends.utils.refresh_user_token', autospec=True)
     def test_student_enrollments_called_task(
-            self, mocked_refresh, mocked_get_enrollments, mocked_get_certificates, mocked_get_current_grades):
+            self, mocked_refresh, mocked_refresh_cache):
         """
         Assert get_student_enrollments is actually called
         """
         batch_update_user_data_subtasks.s(self.students).apply(args=()).get()
-        assert mocked_get_enrollments.called
-        assert mocked_get_certificates.called
-        assert mocked_get_current_grades.called
+        assert mocked_refresh_cache.called
```

Comment:
the problem with asserting this here is that the function is called with an instance of `EdxApi` that is created inside `batch_update_user_data_subtasks`.
Would it be reasonable to make assertions like
`mocked_refresh_cache.assert_any_call(self.user, mock.ANY, 'enrollment')`
basically bypassing the `EdxApi` instance?